aws/sdk/build.gradle.kts

```diff
@@ -314,6 +315,32 @@ tasks.register<ExecRustBuildTool>("hydrateReadme") {
     )
 }
 
+tasks.register<RequireRustBuildTool>("requireCrateHasher") {
+    description = "Ensures the crate-hasher tool is available"
+    inputs.dir(crateHasherToolPath)
+    toolPath = crateHasherToolPath
+}
+
+tasks.register<ExecRustBuildTool>("generateVersionManifest") {
+    description = "Generate the SDK version.toml file"
+    dependsOn("requireCrateHasher")
+    dependsOn("fixManifests")
+
+    inputs.dir(publisherToolPath)
+
+    toolPath = publisherToolPath
+    binaryName = "publisher"
+    arguments = listOf(
+        "generate-version-manifest",
+        "--location",
+        outputDir.absolutePath,
+        "--smithy-build",
+        outputDir.resolve("../../smithy-build.json").normalize().absolutePath,
+        "--examples-revision",
+        properties.get("aws.sdk.examples.revision") ?: "missing"
```

Comment:
will this be confusing when it happens? should we hard fail if this is missing?

Reply:
It's definitely not ideal, but the alternative means not being able to run `./gradlew aws:sdk:assemble` locally without checking out a copy of `aws-doc-sdk-examples`